### PR TITLE
feat: Add get changelog data endpoint

### PR DESCRIPTION
**Pull request by andrcuns**
Add support for fetching changelog data: https://docs.gitlab.com/ee/api/repositories.html#generate-changelog-data

### CODE REVIEW

**Review by NARKOZ — APPROVED** [review]

File: lib/gitlab/client/repositories.rb [hunk 1]
```diff
@@ -109,5 +109,22 @@ def contributors(project, options = {})
     def generate_changelog(project, version, options = {})
       post("/projects/#{url_encode project}/repository/changelog", body: options.merge(version: version))
     end
+
+    # Get changelog data
+    #
+    # @example
+    #   Gitlab.generate_changelog(42, 'v1.0.0')
```
Comment (NARKOZ) [inline lib/gitlab/client/repositories.rb:116]:
```suggestion
    #   Gitlab.get_changelog(42, 'v1.0.0')
```

File: spec/gitlab/client/repositories_spec.rb [hunk 2]
```diff
@@ -132,7 +132,7 @@
 
   describe '.generate_changelog' do
     before do
-      stub_post('/projects/3/repository/changelog', 'changelog')
+      stub_post('/projects/3/repository/changelog', 'generate_changelog')
```
Comment (NARKOZ) [inline spec/gitlab/client/repositories_spec.rb:135]: why was this changed?

**Review by NARKOZ — CHANGES_REQUESTED** [review]
Thanks for the PR. I've left comments.

File: lib/gitlab/client/repositories.rb (hunk 1 above)
Comment (andrcuns) [reply lib/gitlab/client/repositories.rb:116]: Oh, right, missed copy/paste mistake 🤦

File: spec/gitlab/client/repositories_spec.rb (hunk 2 above)
Comment (andrcuns) [reply spec/gitlab/client/repositories_spec.rb:135]: For stub response filenames to better correspond to method names in which test they are used.

BTW, I was the one who also added original `generate_changelog` method.

**Review by NARKOZ — APPROVED** [review]

**Comment by NARKOZ** [comment]
Thank you!
